src/ai/backend/manager/scheduler/predicates.py

```diff
@@ -245,3 +247,76 @@ async def check_domain_resource_limit(
             ),
         )
     return PredicateResult(True)
+
+
+async def check_pending_session_limit(
```

Comment:
How about splitting the pending sesison count limit and the pending resource limit for clarity?